generator/src/test/java/com/scottlogic/deg/generator/fieldspecs/DateTimeRestrictionsMergeOperationTests.java

```diff
@@ -0,0 +1,162 @@
+        DateTimeRestrictionsMerger merger = mock(DateTimeRestrictionsMerger.class);
+        DateTimeRestrictionsMergeOperation operation = new DateTimeRestrictionsMergeOperation(merger);
+        FieldSpec left = FieldSpec.Empty.withDateTimeRestrictions(
+            new DateTimeRestrictions(),
+            FieldSpecSource.Empty);
+        FieldSpec right = FieldSpec.Empty.withDateTimeRestrictions(
+            new DateTimeRestrictions(),
+            FieldSpecSource.Empty);
+        FieldSpec merging = FieldSpec.Empty
+            .withTypeRestrictions(DataTypeRestrictions.createFromWhiteList(IsOfTypeConstraint.Types.NUMERIC), FieldSpecSource.Empty);
+        when(merger.merge(left.getDateTimeRestrictions(), right.getDateTimeRestrictions()))
+            .thenReturn(new MergeResult<>());
+
+        Optional<FieldSpec> result = operation.applyMergeOperation(left, right, merging);
+
+        Assert.assertThat(result.isPresent(), is(true));
+        Assert.assertThat(result.get(), not(sameInstance(merging)));
+        Assert.assertThat(result.get().getDateTimeRestrictions(), is(nullValue()));
+        Assert.assertThat(result.get().getTypeRestrictions(), not(nullValue()));
+        Assert.assertThat(result.get().getTypeRestrictions().getAllowedTypes(), not(hasItem(IsOfTypeConstraint.Types.TEMPORAL)));
+    }
+
+    @Test
+    public void applyMergeOperation_withContradictoryDateTimeRestrictionsAndTemporalTypeOnlyPermittedType_shouldPreventAnyTemporalValues(){
+        DateTimeRestrictionsMerger merger = mock(DateTimeRestrictionsMerger.class);
+        DateTimeRestrictionsMergeOperation operation = new DateTimeRestrictionsMergeOperation(merger);
+        FieldSpec left = FieldSpec.Empty.withDateTimeRestrictions(
+            new DateTimeRestrictions(),
+            FieldSpecSource.Empty);
+        FieldSpec right = FieldSpec.Empty.withDateTimeRestrictions(
+            new DateTimeRestrictions(),
+            FieldSpecSource.Empty);
+        FieldSpec merging = FieldSpec.Empty
+            .withTypeRestrictions(DataTypeRestrictions.createFromWhiteList(IsOfTypeConstraint.Types.TEMPORAL), FieldSpecSource.Empty);
+        when(merger.merge(left.getDateTimeRestrictions(), right.getDateTimeRestrictions()))
+            .thenReturn(new MergeResult<>());
+
+        Optional<FieldSpec> result = operation.applyMergeOperation(left, right, merging);
+
+        Assert.assertThat(result.isPresent(), is(true));
+        Assert.assertThat(result.get(), not(sameInstance(merging)));
+        Assert.assertThat(result.get().getDateTimeRestrictions(), is(nullValue()));
+        Assert.assertThat(result.get().getTypeRestrictions(), not(nullValue()));
+        Assert.assertThat(result.get().getTypeRestrictions().getAllowedTypes(), empty());
+    }
+
+    @Test
+    public void applyMergeOperation_withMergableDateTimeRestrictions_shouldApplyMergedDateTimeRestrictions(){
+        DateTimeRestrictionsMerger merger = mock(DateTimeRestrictionsMerger.class);
```

Comment:
Should you put this `merger`, `operation`, `left` and `right` in a beforeEach, as they are all repeated in every test?